modules/installation-configuring-shiftstack-single-ipv6.adoc

```diff
@@ -0,0 +1,91 @@
+
+You can create a single-stack IPv6 cluster on {rh-openstack-first} after you configure your {rh-openstack} deployment.
+
+IMPORTANT: You cannot convert a dual-stack cluster into a single-stack IPv6 cluster. Additionally, deployments that rely on Fast Datapath are not supported.
+
+.Prerequisites
+
+* Your {rh-openstack} deployment has an existing IPv6 subnet that is DHCPv6-stateful.
+* DNS is configured for the existing IPv6 subnet.
+* The IPv6 subnet is added to a {rh-openstack} router, and the router is configured to send router advertisements (RAs).
+* You added any additional IPv6 subnets that are used in the cluster to an {rh-openstack} router to enable router advertisements.
++
+NOTE: Using an IPv6 SLAAC subnet is not supported due to a known [OpenStack issue](https://bugzilla.redhat.com/show_bug.cgi?id=2304331) that prevents DNS from working.
+* You have a mirror registry with an IPv6 interface.
+* The {rh-openstack} network accepts a minimum MTU size of 1380 bytes.
+* You created API and ingress virtual IP addresses (VIPs) as ports on the network and included those addresses in the `install-config.yaml` file.
+
+.Procedure
+
+. Create the API VIP port on the network by running the following command:
++
+[source,bash]
+----
+$ openstack port create api --network v6-network
+----
+
+. Create the Ingress VIP port on the network by running the following command:
++
+[source,bash]
+----
+$ openstack port create ingress --network v6-network
+----
+
+. After the networking resources are pre-created, deploy a cluster by using an `install-config.yaml` file that reflects your IPv6 network configuration. As an example:
++
+[source,yaml]
+----
+apiVersion: v1
+baseDomain: mydomain.test
+compute:
+- name: worker
+  platform:
+    openstack:
+      type: m1.xlarge
+  replicas: 3
+controlPlane:
+  name: master
+  platform:
+    openstack:
+      type: m1.xlarge
+  replicas: 3
+metadata:
+  name: mycluster
+networking:
+  machineNetwork:
+  - cidr: "fd2e:6f44:5dd8:c956::/64"
+  clusterNetwork:
+  - cidr: fd01::/48
+    hostPrefix: 64
+  serviceNetwork:
+  - fd02::/112
+platform:
+  openstack:
+    ingressVIPs: ['fd2e:6f44:5dd8:c956::383']
+    apiVIPs: ['fd2e:6f44:5dd8:c956::9a']
+    controlPlanePort:
+      fixedIPs: # <1>
+      - subnet:
+          name: subnet-v6
+      network:
+        name: v6-network
+imageContentSources: #<2>
+- mirrors:
+  - <mirror>
+  source: quay.io/openshift-release-dev/ocp-v4.0-art-dev
+- mirrors:
+  - <mirror>
+  source: registry.ci.openshift.org/ocp/release
+additionalTrustBundle: |
+<certificate_of_the_mirror>
+----
+<1> Items under the `platform.openstack.controlPlanePort.fixedIPs` and `platform.openstack.controlPlanePort.network` keys can contain either an ID or a name.
+<2> The image content sources section contains the mirror details. For more information on configuring a local image registry, see "Creating a mirror registry with mirror registry for Red Hat OpenShift".
```

Comment:
Suggestion:
can you highlight that `ingressVIPs` `apiVIPs` must contain the addresses of the ports previously generated? I know you already mentioned before, but maybe pointing out here will be more clear.
Also, the `machineNetwork` CIDR should match the CIDR of the subnet that will be used by the machines which is defined under `controlPlanePort`.

Reply:
In the dual-stack docs we have detailed all fields which are relevant for this type of installation, maybe we could do something similar here?  https://docs.openshift.com/container-platform/4.17/installing/installing_openstack/installing-openstack-installer-custom.html#install-osp-deploy-dualstack_installing-openstack-installer-custom